### PR TITLE
[Android] fix FileUtils::listFiles with path 'assets/'

### PR DESCRIPTION
**Pull request by PatriceJiang**
ref https://github.com/cocos2d/cocos2d-x/issues/19456

### CODE REVIEW

File: cocos/platform/android/CCFileUtils-android.cpp [hunk 1]
```diff
@@ -300,7 +300,7 @@ long FileUtilsAndroid::getFileSize(const std::string& filepath) const
 std::vector<std::string> FileUtilsAndroid::listFiles(const std::string& dirPath) const
 {
 
-    if(isAbsolutePath(dirPath)) return FileUtils::listFiles(dirPath);
+    if(!dirPath.empty() && dirPath[0] == '/') return FileUtils::listFiles(dirPath);
```
Comment (PatriceJiang) [inline cocos/platform/android/CCFileUtils-android.cpp:303]:
```cpp
bool FileUtilsAndroid::isAbsolutePath(const std::string& strPath) const
{
    DECLARE_GUARD;
    // On Android, there are two situations for full path.
    // 1) Files in APK, e.g. assets/path/path/file.png
    // 2) Files not in APK, e.g. /data/data/org.cocos2dx.hellocpp/cache/path/path/file.png, or /sdcard/path/path/file.png.
    // So these two situations need to be checked on Android.
    if (strPath[0] == '/' || strPath.find(_defaultResRootPath) == 0)
    {
        return true;
    }
    return false;
}
```

`_defaultResRootPath` is treated as 'absolute path'

**Comment by imtrobin** [comment]
I applied this line to my code, and it crashes, I think it is infinite stack overflow. If I call it with a path like "/testfolder/", it will be endless loop.

**Comment by PatriceJiang** [comment]
I added two cases 
```cpp
    auto sdcard1 = FileUtils::getInstance()->listFiles ("/sdcard/");
    auto sdcard2 = FileUtils::getInstance()->listFiles ("/sdcard/xxxshoulnotexists");
```
they returns.  
@imtrobin  do I miss something?

**Comment by imtrobin** [comment]
```
if (!fileutil->isAbsolutePath (plistpath))
	plistpath = fileutil->getDefaultResourceRootPath () + "theme/" + plistpath;
vector <string> filelist = fileutil->listFiles (plistpath);

```
My plistpath is "gamecloud/modernlivingroom/animation" so it became 

assets/theme/gamecloud/modernlivingroom/animation

 when I did a debug print. When you see the code, if I pass in "/sdcard/", it should be infinite loop, no?

**Comment by imtrobin** [comment]
sorry, my bad. The above fix works. It found the files, but fullPathForFilename didn't, the crash was happening there. Consider this fixed

**Comment by imtrobin** [comment]
So the windows version will return you the full path, while the android version only returns you the file itself. So when I try to addSpriteFramesWithFile, it is not found on Android, while it works on win32.

**Comment by imtrobin** [comment]
To add on to above, the listfiles returns only the filename for items in /assets, for things in /data folder, it returns the fullpath fine.

**Comment by minggo** [comment]
it is synchronized to metal-support in https://github.com/cocos2d/cocos2d-x/pull/19588

**Pull request merged**